pkg/projects/resource_project.go

```diff
@@ -352,17 +352,18 @@ func projectResource() *schema.Resource {
 
 		data.SetId(project.Id())
 
-		_, err = updateMembers(data.Id(), usersMembershipType, users, m)
+		// Role should be updated first before members or groups as they may depend on roles defined by the users
+		_, err = updateRoles(data.Id(), roles, m)
```

Comment:
Changed the ordering of updating these sub-resources as members and groups can depend on roles created here.